### DISCUSSION TITLE
Quarto reformatting source file – can it be prevented?

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#3102] [Q&A]
I'm a bit confused why Quarto reformats the original source file; can this be turned off? (Or, if I'm missing some good reasons, where can I read about the rationale?)

I noticed it when my definition list,

```markdown
---
title: "test"
format: pdf
editor: visual
---

Item 1:
  ~ this is a line
  ~ this is another

Item 2:
  ~ this is a line
  ~ this is another
  
```

became,

```markdown
---
title: "test"
format: pdf
editor: visual
---

Item 1:

:   this is a line

:   this is another

Item 2:

:   this is a line

:   this is another

```

I prefer the tight syntax above, and clearly it's valid since it was correctly parsed – how can I keep it?

**Comment by mcanouil** [comment]
There seems to be a confusion, when you say "Quarto reformats ...", I don't believe you actually meant that, instead I believe it's "RStudio's visual editor reformats ...".
If you really care about the markdown syntax, don't use the visual editor.

**Reply by baptiste** [reply]
I see, thanks. I hadn't realised it was connected to the visual editor, because I usually turn it off anyway. Since it happened when I reopened the file, I assumed it had been reformatted when I had rendered it, when in fact it's the visual <-> source transition that reformats it. Still, it would be nice to have the transition be a no-op in such cases (but I presume there's a technical advantage in standardising the input format into a subset of the full syntax options).

**Reply by mcanouil** [reply]
If you don't want the visual editor, then disable it in the settings and remove `editor: visual` which tells RStudio to use the visual editor.

**Reply by cscheid** [reply]
I think the answer is, simply, that "maintaining a perfect correspondence between two-way roundtripping visual editor + redundant concrete syntaxes that have the same AST representation is a _very hard_ technical problem".
